### PR TITLE
test: added unit test for updatePodsLoop

### PR DESCRIPTION
**Pull request by smritidahal653**
added unit test for updatePodsLoop in podsTracker.go

### CODE REVIEW

File: pkg/provider/podsTracker_test.go [hunk 1]
```diff
@@ -263,3 +263,81 @@ func TestCleanupDanglingPods(t *testing.T) {
...
+
+	if equality.Semantic.DeepEqual(k8sPods[0], podsCopy[0]) {
+		t.Errorf("pod has been updated so it should not be deeply equal to the original copy")
+	}
+	if equality.Semantic.DeepEqual(k8sPods[1], podsCopy[1]) {
+		t.Errorf("pod has been updated so it should not be deeply equal to the original copy")
+	}
+	if !equality.Semantic.DeepEqual(k8sPods[2], podsCopy[2]) {
+		t.Errorf("pod has not been updated, so it should remain the same as the original copy")
+	}
```
Comment (helayoty) [inline pkg/provider/podsTracker_test.go:332]: We can have this as a for loop

File: pkg/provider/podsTracker_test.go [hunk 2]
```diff
@@ -263,3 +263,81 @@ func TestCleanupDanglingPods(t *testing.T) {
...
+	podName2 := "pod-" + uuid.New().String()
+	podName3 := "pod-" + uuid.New().String()
+	podNamespace := "ns-" + uuid.New().String()
+
+	podsNames := []string{podName1, podName2, podName3}
+	k8sPods := testsutil.CreatePodsList(podsNames, podNamespace)
+
+	k8sPods[0].Status.Phase = v1.PodPending
+	k8sPods[1].Status.Phase = v1.PodRunning
+	k8sPods[2].Status.Phase = v1.PodPending
```
Comment (helayoty) [inline pkg/provider/podsTracker_test.go:278]: May we add the `PodSucceeded ` and `PodFailed ` status phases as well?

Comment (smritidahal653) [reply pkg/provider/podsTracker_test.go:278]: There is a separate unit test for processPodUpdates that tests all the statuses, do we need that in the current unit test as well?

Comment (helayoty) [reply pkg/provider/podsTracker_test.go:278]: Ah, you're right. But I can see the `PodFailed` is covered, but `PodSucceeded` is not. Feel free to add it to the other UT.

**Comment by Fei-Guo** [comment]
LGTM

**Review by helayoty — APPROVED** [review]

**Pull request merged**